h/services/bulk_annotation.py

```diff
@@ -109,36 +109,34 @@ def annotation_search(
     @classmethod
     def _search_query(cls, authority, audience, updated) -> Select:
         """Generate a query which can then be executed to find annotations."""
-        query = sa.select(
-            [cls._AUTHOR.username, Group.authority_provided_id]
-        ).select_from(Annotation)
-
+        query = (
+            sa.select([cls._AUTHOR.username, Group.authority_provided_id]).select_from(
+                AnnotationSlim
+            )
+            # Always join on AUTHOR as we return its username
```

Comment:
Join while creating the query.

This is probably a matter of taste but I prefer this order, first build the query with all necessary joins, then just add filters.